CMakeLists.txt

```diff
@@ -91,6 +92,34 @@ if(ENABLE_ASAN)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address")
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsanitize=address")
```

Comment:
like here, maybe we need to change `CMAKE_C_FLAGS` and `CMAKE_EXE_LINKER_FLAGS`.